oci/casext/gc_test.go

```diff
@@ -0,0 +1,47 @@
+package casext
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/openSUSE/umoci/oci/cas/dir"
+	"golang.org/x/net/context"
+)
+
+func TestGC(t *testing.T) {
+	ctx := context.Background()
+
+	root, err := ioutil.TempDir("", "umoci-TestEngineReference")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(root)
+
+	image := filepath.Join(root, "image")
+	if err := dir.Create(image); err != nil {
+		t.Fatalf("unexpected error creating image: %+v", err)
+	}
+
+	engine, err := dir.Open(image)
+	if err != nil {
+		t.Fatalf("unexpected error opening image: %+v", err)
+	}
+	engineExt := NewEngine(engine)
+	defer engine.Close()
+
+	descMap, err := fakeSetupEngine(t, engineExt)
+	if err != nil {
+		t.Fatalf("unexpected error doing fakeSetupEngine: %+v", err)
+	}
+
+	if descMap == nil {
+		t.Fatalf("empty descMap")
+	}
+
+	err = engineExt.GC(ctx)
+	if err != nil {
+		t.Fatalf("GC failed: %+v", err)
+	}
```

Comment:
Thanks for the test. Should we add something that verifies that GC() actually did the right thing, not just that it didn't return an error?

Reply:
Will look into it. Still learning the code as I go.